apisix/balancer/ewma.lua

```diff
@@ -119,38 +137,76 @@ local function _trans_format(up_nodes)
     return next(peers) and peers or nil
 end
 
-
 local function _ewma_find(ctx, up_nodes)
     local peers
-    local endpoint
 
-    if not up_nodes
-       or core.table.nkeys(up_nodes) == 0 then
+    if not up_nodes or nkeys(up_nodes) == 0 then
         return nil, 'up_nodes empty'
     end
 
-    peers = lrucache_trans_format(ctx.upstream_key, ctx.upstream_version,
-                                  _trans_format, up_nodes)
+    peers = lrucache_trans_format(ctx.upstream_key, ctx.upstream_version, _trans_format, up_nodes)
     if not peers then
         return nil, 'up_nodes trans error'
     end
 
-    if #peers > 1 then
-        endpoint = pick_and_score(peers)
+    local tried_endpoints
+    if not ctx.balancer_tried_servers then
+        tried_endpoints = {}
+        ctx.balancer_tried_servers = tried_endpoints
     else
-        endpoint = peers[1]
+        tried_endpoints = ctx.balancer_tried_servers
     end
 
-    return endpoint.host .. ":" .. endpoint.port
-end
+    local filtered_peers
+    for _, peer in ipairs(peers) do
```

Comment:
ERROR: apisix/balancer/ewma.lua: line 161: getting the Lua global "ipairs"

Reply:
fixed